lib/oli_web/live/community_live/associated/table_model.ex

```diff
@@ -0,0 +1,94 @@
+defmodule OliWeb.CommunityLive.Associated.TableModel do
+  use Surface.LiveComponent
+
+  alias Oli.Groups.CommunityVisibility
+  alias OliWeb.Common.Table.{ColumnSpec, SortableTableModel}
+  alias OliWeb.Router.Helpers, as: Routes
+
+  def get_field(field, association) do
+    case association do
+      %CommunityVisibility{project: project, section: nil} -> Map.get(project, field)
+      %CommunityVisibility{project: nil, section: section} -> Map.get(section, field)
+      association -> Map.get(association, field)
+    end
+  end
+
+  def new(associations, id_field \\ :unique_id, action \\ "select") do
+    action =
+      case action do
+        "select" -> &__MODULE__.render_select_column/3
+        "remove" -> &__MODULE__.render_remove_column/3
+      end
+
+    SortableTableModel.new(
+      rows: associations,
+      column_specs: [
+        %ColumnSpec{
+          name: :title,
+          label: "Title",
+          render_fn: &__MODULE__.render_title_column/3,
+          sort_fn: &__MODULE__.sort_title_column/2
```

Comment:
For now being able to sort only by title, do we want to add the other columns as well?

Reply:
Title seems like a good start IMO